### PR TITLE
SQL: change the size of the list of concrete indices when resolving multiple indices

### LINKED ISSUE #43876: SQL: failure while resolving a set of indices that have fields that are not found in other indices
The simplest example that reproduces this is a minimum of 4 indices where each of them has fields that cannot be found in other indices. Also, this bug is present in 7.2+.

```
POST /index1/_bulk
{"index":{}}
{"field1": "bla"}
```

```
POST /index2/_bulk
{"index":{}}
{"field2": "bla"}
```

```
POST /index3/_bulk
{"index":{}}
{"field3": "bla"}
```

```
POST /index4/_bulk
{"index":{}}
{"field4": "bla"}
```

Then, creating an alias covering all these 4 indices:

```
{
    "actions" : [
    	{ "add" : { "index" : "index4", "alias" : "my_alias" } },
    	{ "add" : { "index" : "index3", "alias" : "my_alias" } },
        { "add" : { "index" : "index2", "alias" : "my_alias" } },
        { "add" : { "index" : "index1", "alias" : "my_alias" } }
    ]
}
```

And running a simple `SELECT * FROM my_alias` will return:

```
{
  "error": {
    "root_cause": [
      {
        "type": "illegal_argument_exception",
        "reason": "Illegal Capacity: -1"
      }
    ],
    "type": "illegal_argument_exception",
    "reason": "Illegal Capacity: -1"
  },
  "status": 400
}
```

### PR DESCRIPTION
**Pull request by astefan**
The size of the ArrayList we were creating was wrong for more than four indices. This PR adds a test covering this scenario, as well.

Fixes https://github.com/elastic/elasticsearch/issues/43876.

### CODE REVIEW

**Comment by astefan** [comment]
@elasticmachine test this please

**Comment by astefan** [comment]
@elasticmachine test this please

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/2

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolver.java [hunk 1]
```diff
@@ -497,7 +497,7 @@ public void resolveAsSeparateMappings(String indexWildcard, String javaRegex, bo
                     if (unmappedIndices.isEmpty() == true) {
                         concreteIndices = asList(capIndices);
                     } else {
-                        concreteIndices = new ArrayList<>(capIndices.length - unmappedIndices.size() + 1);
+                        concreteIndices = new ArrayList<>(capIndices.length);
```
Comment (matriv) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolver.java]: Wouldn't it make sense to have something like `Math.max(1, capIndices.length - unmappedIndices.size())`

**Review by matriv — APPROVED** [review]
LGTM. Left a suggestion

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolver.java (hunk 1 above)
Comment (astefan) [reply x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolver.java]: `capIndices` is the list of indices in which that specific field can be found, while `unmappedIndices` is the list of indices where this specific field cannot be found. And these two come from fieldcaps API. Imo, capIndices and unmappedIndices should have exclusive content and there shouldn't be any relationship between the size of one and the other.

Comment (matriv) [reply x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolver.java]: OK. Got it, thx!

**Pull request merged**